nerfstudio/scripts/downloads/eyeful_tower.py

```diff
@@ -0,0 +1,357 @@
+
+        frames = []
+        for i in indices:
+            frames.append(transforms["frames"][i])
+
+        output = copy.deepcopy(transforms)
+        output["frames"] = frames
+
+        # Remove the unused files from the splits
+        filenames = {f["file_path"] for f in frames}
+        for key in ["train_filenames", "val_filenames"]:
+            output[key] = sorted(list(set(transforms[key]) & filenames))
+
+        return output
+
+    def download(self, save_dir: Path):
+        if len(self.capture_name) == 0:
+            self.capture_name = ("riverview",)
+            print(
+                f"No capture specified, using {self.capture_name} by default.",
+                "Add `--help` to this command to see all available captures.",
+            )
+
+        if len(self.resolution_name) == 0:
+            self.resolution_name = ("jpeg_2k",)
+            print(
+                f"No resolution specified, using {self.resolution_name} by default.",
+                "Add `--help` to this command to see all available resolutions.",
+            )
+
+        captures = set()
+        for capture in self.capture_name:
+            if capture == "all":
+                captures.update([c for c in eyefultower_downloads if c != "all"])
+            else:
+                captures.add(capture)
+        captures = sorted(captures)
+        if len(captures) == 0:
+            print("WARNING: No EyefulTower captures specified. Nothing will be downloaded.")
+
+        resolutions = set()
+        for resolution in self.resolution_name:
+            if resolution == "all":
+                resolutions.update([r for r in eyefultower_resolutions.keys() if r != "all"])
+            else:
+                resolutions.add(resolution)
+        resolutions = sorted(resolutions)
+        if len(resolutions) == 0:
+            print("WARNING: No EyefulTower resolutions specified. Nothing will be downloaded.")
+
+        driver = awscli.clidriver.create_clidriver()
+
+        for i, capture in enumerate(captures):
+            base_url = f"s3://fb-baas-f32eacb9-8abb-11eb-b2b8-4857dd089e15/EyefulTower/{capture}/"
+            output_path = save_dir / "eyefultower" / capture
+            includes = []
+            for resolution in resolutions:
+                includes.extend(["--include", f"{eyefultower_resolutions[resolution].folder_name}/*"])
+            command = (
+                ["s3", "sync", "--no-sign-request", "--only-show-errors", "--exclude", "images*/*"]
+                + includes
+                + [base_url, str(output_path)]
+            )
+            print(f"[EyefulTower Capture {i+1: >2d}/{len(captures)}]: '{capture}'")
+            print(
+                f"    Downloading resolutions {resolutions}",
+                f"to '{output_path.resolve()}' with command `aws {' '.join(command)}` ...",
+                end=" ",
+                flush=True,
+            )
+            driver.main(command)
+            print("done!")
```

Comment:
Same here, wrt. printing.

Reply:
This one (and the previous one) I've actually left as `print` since `CONSOLE.print` doesn't have a `flush=True` parameter. I'm using the `flush` argument to indicate what's going on without starting a new line (which is done with `done!` output).

Reply:
Sounds good!